From 2237391e979b2423462bb5eaf1fdf4c0876f7476 Mon Sep 17 00:00:00 2001
From: Xaver Hugl <xaver.hugl@gmail.com>
Date: Wed, 10 May 2023 12:28:24 +0200
Subject: [PATCH] xcbutils: fix nativeFloor

The floor was not done in the X-native coordinate space but with double
scaling

BUG: 459373


(cherry picked from commit a48c98f5aba71fab8bc1e1d5dd18668833631920)
---
 src/utils/xcbutils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/xcbutils.cpp b/src/utils/xcbutils.cpp
index cea0ef7c894..9731f483c0b 100644
--- a/src/utils/xcbutils.cpp
+++ b/src/utils/xcbutils.cpp
@@ -646,7 +646,7 @@ QSizeF fromXNative(const QSize &s)
 
 static qreal nativeFloor(qreal value)
 {
-    return std::floor(value / kwinApp()->xwaylandScale()) * kwinApp()->xwaylandScale();
+    return std::floor(value * kwinApp()->xwaylandScale()) / kwinApp()->xwaylandScale();
 }
 
 QRectF nativeFloor(const QRectF &rect)
-- 
GitLab

